### ISSUE TITLE
config IsEnabled from false to true

### ISSUE THREAD

**Issue by phoeson** [nortal/Utilities.TaskSchedulerEngine#2]
It's not able to config a task from IsEnabled false to true while running

**Comment by ImrePyhvel** [comment]
Yes, the current implementation is not too responsive about `ISchedulerTask.IsEnabled` nor `ISchedulerTask.Interval` not `ISchedulerTask.ExecutionTime`. **It was initially designed and tested for "configure first, execute second then forget"-usage.** 

Still, it does react to runtime configuration change, but with some limitations, see below.

### Current behavior
The `SchedulerEngineBase` is currently looking at your configuration (including if the task is enabled) at the time the task is inserted to schedule. That is:
* on engine startup when initial schedule is created
* after the task was executed (to add the next run to schedule).

It does not, however, take notice of `IsEnabled` change while it is sitting in the schedule waiting it's turn. 
When you enable previously disabled task, it's currently not updated in the schedule. since the engine is not polling the task configuration. Unfortunately this is the current design.

On the flip side, when you disable previously enabled task it runs just one more time and then stops. I do consider this a bug that should be fixed.

### Fix considerations

To avoid that one excess run when disabling could be fixed by adding a condition before actual task execution. 

To support the runtime enabling of a task or support other configuration parameters being effective immediately would be a bit trickier and require more changes (i.e. constantly recalculate schedule or redesign how the next run time is defined). There would be some tradeoffs I'm not sure would be worthy of it. I do suggest you take the workaround for the time being instead.

### Workaround
The best workaround available right now is to stop and restart the the service if your configuration changes. You don't have to tear down you host process, just Stop/Starting the engine would suffice.
